packages/plugin-compiler/src/main/kotlin/io/realm/kotlin/compiler/Registrar.kt

```diff
@@ -56,14 +54,22 @@ class Registrar : ComponentRegistrar {
             configuration.get(CLIConfigurationKeys.MESSAGE_COLLECTOR_KEY, MessageCollector.NONE)
         SchemaCollector.properties.clear()
 
-        // Trigger generation of companion objects and addition of the RealmObjectCompanion to it
-        SyntheticResolveExtension.registerExtension(
-            project,
-            RealmModelSyntheticCompanionExtension()
-        )
-
-        // Adds RealmObjectInternal properties, rewires accessors and adds static companion
-        // properties and methods
-        IrGenerationExtension.registerExtension(project, RealmModelLoweringExtension())
+        // We load our extensions LAST to avoid exposing our internal attributes to other plugins,
+        // like serialization (issue https://github.com/realm/realm-kotlin/issues/1251).
+        project.extensionArea.run {
+            // Trigger generation of companion objects and addition of the RealmObjectCompanion to it
+            getExtensionPoint(SyntheticResolveExtension.extensionPointName).registerExtension(
+                RealmModelSyntheticCompanionExtension(),
+                LoadingOrder.LAST,
```

Comment:
What happens if multiple plugins define this? I assume behaviour among those is undefined, which I guess is fine. But could users also do this by just reorder the plugins in their build file?

Reply:
I wasn't able to see a change by reordering the plugins, neither before or after this update. Only other observation that I can chime in with is that I tried I guess it would be best if we could somehow be order agnostic. I tried adding the `@Transient` annotation to the internal attributes from our compiler plugin and that brings compilation past the original error of not being able to find the serializer for `RealmObjectReference` but then compilation fails with 
```
Caused by: java.lang.IllegalStateException: Function has no body: FUN name:write$Self visibility:public modality:FINAL <> (self:com.example.github1252.CityDto, output:kotlinx.serialization.encoding.CompositeEncoder, serialDesc:kotlinx.serialization.descriptors.SerialDescriptor) returnType:kotlin.Unit
```
I fail to see why we should mess up the generated body for that, but maybe it is because we add the supertypes in the lowering instead of fully syntheziced types. And this somehow all goes away if we just load our extensions last, so ... I went with that for now.